c/driver/postgresql/statement.cc

```diff
@@ -797,9 +664,33 @@ AdbcStatusCode PostgresStatement::ExecuteQuery(struct ArrowArrayStream* stream,
       PQclear(result);
       return ADBC_STATUS_IO;
     }
-    AdbcStatusCode status = InferSchema(*type_resolver_, result, &reader_.schema_, error);
+
+    // Resolve the information from the PGresult into a PostgresType
+    PostgresType root_type;
+    AdbcStatusCode status =
+        ResolvePostgresType(*type_resolver_, result, &root_type, error);
     PQclear(result);
     if (status != ADBC_STATUS_OK) return status;
+
+    // Initialize the copy reader and infer the output schema (i.e., error for
+    // unsupported types before issuing the COPY query)
+    reader_.copy_reader_.reset(new PostgresCopyStreamReader());
```

Comment:
Use make_unique?

Reply:
Really? I was using C++17...

Reply:
What is the exact date for R? 

I suppose this is going to push me to default to Go drivers by default for everything, since at least we get a bit of sanity there.

I'll fix this: https://github.com/apache/arrow-adbc/blob/3c0ab88885d21c2e2582ba0a82ce1bbb7bb2f492/c/cmake_modules/AdbcDefines.cmake#L24

Reply:
Ah, fun.

Well, we'll see how things go. I do plan on getting an ODBC wrapper up when I can, and possibly a JDBC bridge as well. Those will likely be C/C++ (since I plan on scavenging turbodbc), though there is possibility of Rust (since the Rust Arrow projects have some interesting ODBC code).

And of course Rust is coming down the pipeline with Will's work.